qiskit/compiler/schedule.py

```diff
@@ -70,9 +82,19 @@ def schedule(circuits: Union[QuantumCircuit, List[QuantumCircuit]],
         if backend is None:
             raise QiskitError("Must supply either a backend or a meas_map for scheduling passes.")
         meas_map = backend.configuration().meas_map
+    if instruction_durations is None:
+        if backend is None:
+            raise QiskitError("Must supply either a backend or a instruction_durations for "
+                              "scheduling passes.")
+        instruction_durations = InstructionDurations.from_backend(backend)
+    if dt is None:
+        if backend is not None:
+            dt = backend.configuration().dt
 
-    schedule_config = ScheduleConfig(inst_map=inst_map, meas_map=meas_map)
+    schedule_config = ScheduleConfig(inst_map=inst_map, meas_map=meas_map, dt=dt)
     circuits = circuits if isinstance(circuits, list) else [circuits]
+    for circ in circuits:
+        circ.instruction_durations = instruction_durations
```

Comment:
this is only needed for `method='sequence'` right? and if it's another method, then this can be None